EmpowerPlant/EmpowerPlantViewController.swift

```diff
@@ -31,36 +33,125 @@ class EmpowerPlantViewController: UIViewController, UITableViewDelegate, UITable
         configureNavigationItems()
 
         /* TODO
          1 get products from server (so we get http.client span)
          2 check if any products in Core Data -> If Not -> insert the products from response into Core Data
          3 get products from DB (so we get db.query span) and reload the table with this data
          */
+        
         getAllProductsFromServer()
         getAllProductsFromDb()
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return products.count
+    @objc
+    func addToDb() {
+        let alert = UIAlertController(title: "New Product",
+                                      message: "Enter new product title",
+                                      preferredStyle: .alert)
+        alert.addTextField(configurationHandler: nil)
+        
+        alert.addAction(UIAlertAction(title:"Submit", style: .cancel, handler: { [weak self] _ in
+            guard let field = alert.textFields?.first, let text = field.text, !text.isEmpty else {
+                return
+            }
+            self?.createProduct(productId: "123", title: text, productDescription: "product.description", productDescriptionFull: "product.description.full", img:"img", imgCropped:"img.cropped", price:"1")
+        }))
+        
+        self.present(alert, animated: true, completion: nil)
+        
+        // ALSO WORKED
+        // alert.addTextField()
+        // let submitButton = UIAlertAction(title:"Add", style: .default) { (action) in
+        //     print("here")
+        //     let textfield = alert.textFields![0]
+        // }
+        // alert.addAction(submitButton)
+        // self.present(alert, animated: true, completion: nil)
     }
     
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let model = products[indexPath.row]
-        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
-        cell.textLabel?.text = model.title
-        return cell
+    // Don't deprecate, this function is useful for development and testing
+    @objc
+    func clearDb() {
+        print("> clearDb")
+        // self.products was already set by viewDidLoad()
+        // self.products = try context.fetch(Product.fetchRequest())
+        for product in self.products {
+            deleteProduct(product: product)
+        }
+        refreshTable()
+    }
+    
+    private func configureNavigationItems() {
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(
+            title: "Cart",
+            style: .plain,
+            target: self,
+            action: #selector(goToCart) // addToDb
+        )
+        
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(
+            title: "List App",
+            style: .plain,
+            target: self,
+            action: #selector(goToListApp) // clearDb
+        )
+    }
+    
+    // Writes to CoreData database
```

Comment:
@thinkocapo can you remind me why we're creating products in the iOS database instead of fetching products from one of the existing backends that have their own databases? I was assuming mobile would be treated like a frontend/client application in the same way our react app is, so wanted to know if there is a reason for using the DB in the iOS app itself?

Reply:
We're doing it because the sentry-cocoa sdk captures the db queries as spans for the transactions.

It kind of emulates caching, saving state/stuff locally rather than having to fetch. Not fully realistic but it's better than nothing or writing dummy data queries. 

If user navigates away, but then comes back, and has since lost wifi, then no http call is needed, because the data was saved locally.

@cstavitsky

Reply:
But it also fetches the products from the backend, so we get a span for that. If the database is empty, then it writes them to the database. But need db query + http request every single time.

The point is to get spans. The business logic and what's going on is opaque to the user/audience.